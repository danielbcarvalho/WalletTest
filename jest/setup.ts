import 'react-native-gesture-handler/jestSetup';
import { mockDefaultTheme } from '../src/theme/defaultTheme';

jest.mock('styled-components', () => {
  const original = jest.requireActual('styled-components');

  return {
    ...original,
    useTheme: () => {
      return mockDefaultTheme;
    },
  };
});
