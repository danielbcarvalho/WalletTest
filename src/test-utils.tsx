// TODO: Verificar essa melhoria nos testes
import React from 'react';
import {
  render,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { mockDefaultTheme } from './theme/defaultTheme';

interface IAllThemeProviders {
  children: React.ReactElement;
}
const AllTheProviders = ({ children }: IAllThemeProviders) => {
  return <ThemeProvider theme={mockDefaultTheme}>{children}</ThemeProvider>;
};

function customRender(
  ui: React.ReactElement,
  options = {} as RenderOptions,
): RenderAPI {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

export function rendererCreate(
  component: React.ReactElement,
): ReactTestRenderer {
  return renderer.create(<AllTheProviders>{component}</AllTheProviders>);
}

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
