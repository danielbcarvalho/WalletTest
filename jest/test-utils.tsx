import React, { ReactElement, FC } from 'react';
import { render } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ThemeProvider } from '../src/theme';

type Options = Parameters<typeof render>[1];

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: Infinity,
    },
  },
});
const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
