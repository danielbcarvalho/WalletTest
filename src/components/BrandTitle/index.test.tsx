import React from 'react';

import { render } from '../../../jest/test-utils';

import StyledBackgroundContainer from '.';

describe('StyledBackgroundContainer', () => {
  test('matches snapshot', () => {
    const { toJSON } = render(<StyledBackgroundContainer />);
    expect(toJSON()).toMatchSnapshot();
  });
});
