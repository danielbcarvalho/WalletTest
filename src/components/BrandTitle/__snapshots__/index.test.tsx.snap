// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledBackgroundContainer matches snapshot 1`] = `
<Text
  style={
    {
      "color": "#FFF",
      "fontFamily": "PTSans-Caption",
      "fontSize": 28,
      "lineHeight": 52,
    }
  }
>
  Wallet Test
</Text>
`;
